x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/allocation/TrainedModelAllocationClusterService.java

```diff
@@ -215,8 +215,12 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         });
     }
 
-    private static ClusterState update(ClusterState currentState, TrainedModelAllocationMetadata.Builder modelAllocations) {
-        if (modelAllocations.isChanged()) {
+    private static ClusterState update(
+        ClusterState currentState,
+        TrainedModelAllocationMetadata.Builder modelAllocations,
+        boolean isChanged
```

Comment:
Would this be better captured if it was named `force` instead of `isChanged`? It seems like it's effectively forcing the update even if there are no changes in `modelAllocation`.